downstairs/src/lib.rs

```diff
@@ -1922,10 +2049,14 @@ impl Downstairs {
                     result,
                 }))
             }
-            IOop::Write {
-                dependencies,
-                writes,
-            } => {
+            IOop::Write { .. } => {
+                let IOop::Write {
+                    writes,
+                    dependencies,
+                } = job.work
+                else {
+                    unreachable!();
+                };
```

Comment:
what does this change (and the similar one above it) do? Is this something to make the rust compiler happy?

Reply:
This was intentional at the time: the `match` borrowed `job.work`, and then this statement transferred ownership in only particular cases.  However, it's not necessary; see https://github.com/oxidecomputer/crucible/pull/1272#discussion_r1600406065